src/executor/executor.ts

```diff
@@ -385,11 +390,29 @@ export class TaskExecutor {
       timeout: options?.timeout ?? this.options.taskTimeout,
     });
     this.taskQueue.addToEnd(task as Task<unknown, unknown>);
+    if (!this.startupTimeoutError && !this.startupTimeoutId) {
```

Comment:
I have few issues with this, we can talk about this f2f:

- This way of communicating the error with the timeout isn't easy for the user to handle, we should work on that
- Since this is implemented as part of `executeTask`, this timeout will be started and counted for each scheduled task. I do see a concurrency problem here, given all the tasks are executed on the same TaskExecutor instance.

Reply:
this timeout will only start once, unless `this.startupTimeoutId` exists yet, so there shouldn't be a problem with concurrency, I think, but let's talk about it..